src/chain/chain.rs

```diff
@@ -371,9 +371,19 @@ impl Chain {
         // Check the lowest version of our info that any neighbour has and remove everything less
         // than it
         let our_oldest_ver = self.their_knowledge.values().min().map_or(0, |&v| v);
-        // TODO since the list is sorted according to the version, choose something more efficient
-        self.our_infos
-            .retain(|&(ref si, _)| *si.version() >= our_oldest_ver);
+        let oldest_ver_index = self
+            .our_infos
+            .binary_search_by_key(&our_oldest_ver, |&(ref si, _)| *si.version())
```

Comment:
Minor: you can simplify the lambda params to just `|(si, _)|` due to match ergonomics - the compiler can figure the `ref`s and `&`s on its own.

Reply:
Done